### PR TITLE
Add: login form files

### PR DESCRIPTION
**Pull request by PavanVaranasi2**
- Login form includes two textfields with icons and a login button
- on clicking login button the details are verified with already existing users information in dummyUsers file.
- At present remember me and forgot password is not having any functionality

Screenshot 2024-06-19 at 12 54 41 PM
Screenshot 2024-06-19 at 12 55 02 PM
Screenshot 2024-06-19 at 12 55 19 PM
Screenshot 2024-06-19 at 12 55 36 PM

### CODE REVIEW

File: src/components/molecules/LoginForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const LoginForm: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [remember, setRemember] = useState(false);
+  const [error, setError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleLogin = () => {
+    if (!email || !password) {
+      setError('Email and Password are required');
```
Comment (MajaharZemoso) [inline src/components/molecules/LoginForm/index.tsx]: We are not allowed to add functionalities in molecule level
we are supposed to add them in organism level

File: src/components/molecules/LoginForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+  };
+
+  const toggleShowPassword = () => {
+    setShowPassword((prev) => !prev);
+  };
+
+  return (
+    <Box 
+      component="form"
```
Comment (MajaharZemoso) [inline src/components/molecules/LoginForm/index.tsx]: Don't do inline stylings 
we have styled component in mui you can use that

File: src/components/templates/LoginTemplate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// src/components/templates/LoginTemplate.tsx
+import React from 'react';
+import Login from '../../organisms/Login';
+import { Container } from '@mui/material';
+
+const LoginTemplate: React.FC = () => {
+  return (
+    <Container maxWidth="sm">
+      <Login />
```
Comment (MajaharZemoso) [inline src/components/templates/LoginTemplate/index.tsx]: Don't add organism in template

Note : revise Atomic Structure once

**Review by PavanVaranasi2 — COMMENTED** [review]
Refactor LoginForm to adhere to Atomic Design principles

- **Functional Isolation**: All functional logic has been removed from 'molecules' and is now available only in 'organisms'. 

- **Introduction of Styled API**: Implemented MUI's `styled` API across all components, replacing inline styles.

- **Refactor of Molecules**: Molecules such as `LoginForm` have been stripped of direct logic like event handling and state management

Screenshot 2024-06-21 at 1 27 39 PM
Screenshot 2024-06-21 at 1 28 10 PM
Screenshot 2024-06-21 at 1 28 26 PM

File: src/components/organisms/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  textAlign: 'center'  // Ensures text is centered if Typography width is not full
+}));
+
+const Login: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [remember, setRemember] = useState(false);
+  const [error, setError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [successMessage, setSuccessMessage] = useState('');
```
Comment (MajaharZemoso) [inline src/components/organisms/Login/index.tsx]: add types to useState

File: src/components/templates/LoginTemplate/index.tsx (hunk 3 above)
Comment (MajaharZemoso) [reply src/components/templates/LoginTemplate/index.tsx]: This is not resolved yet

**Review by PavanVaranasi2 — COMMENTED** [review]
- removed direct usage of organism in template and used it in pages instead
- Also added types to useState variables in organisms

**Review by MajaharZemoso — APPROVED** [review]

**Pull request merged**